starport_template/lib/pages/wallet_details_page.dart

```diff
@@ -1,14 +1,109 @@
+class WalletDetailsPage extends StatefulWidget {
+  final WalletInfo walletInfo;
+
+  const WalletDetailsPage({Key? key, required this.walletInfo}) : super(key: key);
+
+  @override
+  _WalletDetailsPageState createState() => _WalletDetailsPageState();
+}
+
+class _WalletDetailsPageState extends State<WalletDetailsPage> {
+  Observable<List<Balance>>? get balancesList => StarportApp.walletsStore.balancesList;
+
+  Observable<bool> get isBalancesLoading => StarportApp.walletsStore.isBalancesLoading;
+
+  Observable<bool> get isSendMoneyLoading => StarportApp.walletsStore.isSendMoneyLoading;
+
+  Observable<bool> get isError => StarportApp.walletsStore.isError;
+
+  @override
+  void initState() {
+    super.initState();
+    _fetchWalletBalances();
+  }
 
   @override
   Widget build(BuildContext context) {
-    return const Scaffold(
-      body: Center(
-        child: Text("Wallet Details Page"),
+    return Scaffold(
+      body: SafeArea(
+        child: Center(
+          child: ContentStateSwitcher(
+            contentChild: Column(
+              children: [
+                ListTile(
+                  title: const Text('Wallet address'),
+                  subtitle: Text(widget.walletInfo.address),
+                ),
+                const Divider(),
+                const Padding(padding: EdgeInsets.only(top: 16)),
+                BalanceHeading(),
+                if (balancesList != null)
+                  Padding(
+                    padding: const EdgeInsets.all(8.0),
+                    child: Column(
+                      children: balancesList!.value
+                          .map(
+                            (balance) => BalanceCard(
+                              denomText: balance.denom.text,
+                              amountDisplayText: balance.amount.value.toString(),
+                              onTransferPressed: () => _transferPressed(balance),
+                            ),
+                          )
+                          .toList(),
+                    ),
+                  ),
+                if (isSendMoneyLoading.value)
+                  const Padding(
+                    padding: EdgeInsets.only(top: 8.0),
+                    child: Center(
+                      child: Text(
+                        'Sending money',
+                        textAlign: TextAlign.center,
+                      ),
+                    ),
+                  ),
+              ],
+            ),
+            isLoading: isBalancesLoading.value,
+            isError: isError.value,
+            errorChild: const Center(
+              child: Text('An unexpected error occurred'),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  void _transferPressed(Balance balance) {
+    final denom = Denom(balance.denom.text);
+    _openSendMoneySheet(denom);
+  }
+
+  Future _fetchWalletBalances() async {
+    await StarportApp.walletsStore.getBalances(widget.walletInfo.address);
+    setState(() {});
```

Comment:
i just realized we dont need this setState here since we're not updating anything and use mobx under the hood